### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by HeDo88TH**
Closes https://github.com/DroneDB/Registry/issues/173 
Closes https://github.com/DroneDB/Registry/issues/172 
Closes https://github.com/DroneDB/Registry/issues/167

Still working on https://github.com/DroneDB/Registry/issues/165

### CODE REVIEW

**Comment by HeDo88TH** [comment]
There is no easy way to close https://github.com/DroneDB/Registry/issues/165 because we are abstracting the underlying storage system. If this feature becomes critical I'll get back to it but It basically means that we need a vertical integration to make it work efficiently. I added support for range processing when requesting a single file, it should be enough by now.

**Comment by pierotofy** [comment]
> it should be enough by now. 

Yes it actually should be enough; the biggest use case is efficient requests of Cloud Optimized GeoTIFFs, which require accept-ranges to be fetched efficiently over the wire.

Awesome :clap:

File: Registry.Web/Services/Managers/ObjectsManager.cs [hunk 1]
```diff
@@ -501,19 +501,30 @@ public async Task<FileDescriptorDto> GenerateTile(string orgSlug, string dsSlug,
...
+                };
+            }
+            catch (InvalidOperationException ex)
             {
-                ContentStream = memory,
-                ContentType = "image/png",
-                Name = fileName
-            };
+                // NOTE: This is the definition of self-inflicted wound
+                if (ex.InnerException != null && ex.InnerException.Message.Contains("Out of bounds", StringComparison.OrdinalIgnoreCase))
```
Comment (pierotofy) [inline Registry.Web/Services/Managers/ObjectsManager.cs:523]: :laughing:

**Review by pierotofy — APPROVED** [review]

File: Registry.Web/Services/Managers/ObjectsManager.cs (hunk 1 above)
Comment (HeDo88TH) [reply Registry.Web/Services/Managers/ObjectsManager.cs:523]: No one I know told me to do that 😁

**Pull request merged**